src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.RenameFileEditor.cs

```diff
@@ -29,18 +29,18 @@ private IEnumerable<CodeActionOperation> RenameFileToMatchTypeName()
                 var solution = SemanticDocument.Document.Project.Solution;
                 var text = SemanticDocument.Text;
                 var oldDocumentId = SemanticDocument.Document.Id;
-                var newDocumentId = DocumentId.CreateNewId(SemanticDocument.Document.Project.Id, State.TargetFileNameCandidate);
+                var newDocumentId = DocumentId.CreateNewId(SemanticDocument.Document.Project.Id, FileName);
 
                 // currently, document rename is accomplished by a remove followed by an add.
                 // the workspace takes care of resolving conflicts if the document name is not unique in the project
                 // by adding numeric suffixes to the new document being added.
                 var newSolution = solution.RemoveDocument(oldDocumentId);
-                newSolution = newSolution.AddDocument(newDocumentId, State.TargetFileNameCandidate, text);
+                newSolution = newSolution.AddDocument(newDocumentId, FileName, text);
 
                 return new CodeActionOperation[]
                 {
                     new ApplyChangesOperation(newSolution),
-                    new OpenDocumentOperation(newDocumentId)
+                    new OpenDocumentOperation(newDocumentId, activateIfAlreadyOpen: true)
```

Comment:
Thats right. I'll think about it. the navigating to span operation must be wrapped as a codeaction operation. I'll address this in a different PR.